### PR TITLE
Refactor color classes

### PR DESCRIPTION
**Pull request by ChloeDawn**
- `render/block/BlockColorMapper` -> `color/block/BlockColorProvider`
- `render/block/BlockColorMap` -> `color/block/BlockColors`
  - `getRenderColor` -> `getColor` (returns material color if no multiplier present)
  - `getColorMultiplier` (returns -1 if no multiplier present)
- `render/item/ItemColorMapper` -> `color/item/ItemColorProvider`
- `render/item/ItemColorMap` -> `color/item/ItemColors`
  - `getRenderColor` -> `getColorMultiplier` (returns -1 if no multiplier present)
- `render/block/BiomeColors` -> `color/world/BiomeColors`
  - `$Proxy` -> `$Provider`
  - `colorAt` -> `getColor`
  - `foliageColorAt` -> `getFoliageColor`
  - `grassColorAt` -> `getGrassColor`
  - `waterColorAt` -> `getWaterColor`
- `render/block/FoliageColorHandler` -> `color/world/FoliageColors`
- `render/block/GrassColorHandler` -> `color/world/GrassColors`

### CODE REVIEW

**Comment by NeunEinser** [comment]
You keep asking on Discord for a review, but I don't have any context on this and I do not know why the renames were done and I cannot judge whether this might be a good or a bad change.

**Comment by modmuss50** [comment]
the new names seem fine to me, but ill leave this open for review to someone else who knows this area a bit better.

**Review by Runemoro — APPROVED** [review]

File: mappings/net/minecraft/client/color/world/BiomeColors.mapping [hunk 1]
```diff
@@ -0,0 +1,28 @@
+CLASS dka net/minecraft/client/color/world/BiomeColors
+	CLASS dka$a Provider
+		METHOD getColor (Lbie;Lev;)I
+			ARG 1 biome
+			ARG 2 pos
+	FIELD a GRASS_COLOR Ldka$a;
+	FIELD b FOLIAGE_COLOR Ldka$a;
+	FIELD c WATER_COLOR Ldka$a;
+	FIELD d WATER_FOG_COLOR Ldka$a;
+	METHOD a getGrassColor (Lbgp;Lev;)I
```
Comment (Gegy) [inline mappings/net/minecraft/client/color/world/BiomeColors.mapping:10]: What are thoughts on `At` suffix for these?

Comment (ChloeDawn) [reply mappings/net/minecraft/client/color/world/BiomeColors.mapping:10]: The suffix would be redundant, and conflict with Yarn's (and Java's) current naming conventions. An `At` suffix would be something for another PR if it is something you are interested in - it is out of scope for this PR, and changes would need to be reflected across the entire codebase with methods such as `getBlockState`/`getBiome`/`getBlockEntity` for naming consistency.

Comment (ChloeDawn) [reply mappings/net/minecraft/client/color/world/BiomeColors.mapping:10]: Essentially, the positional behaviour of the method is already implied by the `BlockPos` parameter, much like an indexing behaviour is implied through the use of `List#get(int)`

Comment (Gegy) [reply mappings/net/minecraft/client/color/world/BiomeColors.mapping:10]: Ah, okay. That does make sense. Thanks

**Review by Gegy — APPROVED** [review]

**Review by modmuss50 — APPROVED** [review]

**Pull request merged**